src/v2/providers/schedule.ts

```diff
@@ -0,0 +1,198 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2022 Firebase
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import * as express from 'express';
+import { logger } from '../..';
+import { copyIfPresent } from '../../common/encoding';
+import { timezone } from '../../common/timezone';
+import { ManifestEndpoint, ManifestRequiredAPI } from '../../runtime/manifest';
+import * as options from '../options';
+import { HttpsFunction } from './https';
+
+/** @hidden */
+interface ScheduleArgs {
+  schedule: string;
+  timeZone?: timezone;
+  retryCount?: number;
+  maxRetryDuration?: string;
+  minBackoffDuration?: string;
+  maxBackoffDuration?: string;
```

Comment:
Not the best time to have this discussion, but I'm surprised that this is `*Duration` instead of `*Seconds`. That's what we did for Task Queue Functions (even though the underlying Tasks API uses `*Duration`).

On one hand, I'd like consistency across v2. On the other hand, this breaks consistency w/ v1 scheduled function! What do you think we should do?

Reply:
Hmm that's an interesting point. A counter argument is that the user could have a string value that's not in seconds. The API uses `1h` as an example in this case. IDK if we should name the interface something that could limit the customers choices. But on the other hand, we try to have an opinionated SDK.  🤷‍♂️ 

Also, I'm not sure what you mean by breaking consistency w/ v1 scheduled functions since they use the `Duration` naming too.

Reply:
Meant that `*Seconds` breaks consistency w/ Task Queue API. Lots of parameters here look like TQ API's parameter - I would find it weird to see TQ and Scheduled function use slightly different parameters for configuration.

Reply:
After internal discussion, I'll change from *Duration to *Seconds for the v2 api

Reply:
I'd vote for keeping v1 the same.